crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -379,4 +438,50 @@ mod tests {
         assert!((0.95..=1.05).contains(&usdc_price.unwrap()));
         assert_eq!(nonsense_price, Err(PriceEstimationError::NoLiquidity));
     }
+
+    #[tokio::test]
+    #[ignore]
+    async fn prices_adjusted_for_token_decimals() {
+        let usdc = addr!("2a22f9c3b484c3629090FeED35F17Ff8F88f76F0");
+        let wxdai = addr!("e91D153E0b41518A2Ce8Dd3D7944Fa863463a97d");
+        let mut mock = MockTokenInfoFetching::new();
+        mock.expect_get_token_infos().returning(move |tokens| {
+            tokens
+                .iter()
+                .map(|t| {
+                    let decimals = if *t == usdc { Some(6) } else { Some(18) };
```

Comment:
is it possible that a token has more than 18 decimals? if yes, could you please add a token like that in that test to check that the reverse operation (powi with a negative number + then multiplication to normalize it) works as expected? :pray:

Reply:
Yeah, good idea. This was actually a problem in the original fix for the oneinch estimator adjustment since we used `u64` operations there and `10 ** 21` overflowed that value which caused other issues.